src/processes.c

```diff
@@ -1526,15 +1526,15 @@ static char *ps_get_owner(pid_t pid)
         if (strncmp (line, "Uid:", 4) != 0)
             continue;
 
-        uid = strtoul (line + 5);
+        uid = strtoul (line + 5, /* endptr*/ NULL, /* base */ 10);
         getpwuid_r (uid, &passwd, passwd_buffer, sizeof(passwd_buffer),
                 &passwd_result);
         if (passwd_result) {
             result = sstrdup (passwd_result->pw_name);
         } else {
 	    // Send the numeric uid if name is not available.
             char uid_str[21];  // Worst case is 20 digits plus string terminator.
-            snprintf(uid_str, sizeof(uid_str), "%llu", uid);
+            snprintf(uid_str, sizeof(uid_str), "%lu", uid);
```

Comment:
Not quite unfortunately. The issue here is that the format specifier needs to match the type of the argument. ```%lu``` needs an ```unsigned long```, but you are passing a ```uid_t``` (which may or may not be an unsigned long).  I think the shortest way to deal with this is

```
snprintf(uid_str, sizeof(uid_str), "%lu", (unsigned long)uid);
```